src/main/java/com/jfrog/ide/idea/scan/PypiScanManager.java

```diff
@@ -0,0 +1,211 @@
+        }
+    }
+
+    /**
+     * Refresh Python SDKs and listen to changes in the SDK environment.
+     * For example, a scan should be triggered after running 'pip install'.
+     */
+    void refreshPythonSdks() {
+        if (!Disposer.isDisposed(sdkDisposable)) {
+            // Remove old "RunOnChangeUnderInterpreterPaths" listeners
+            Disposer.dispose(sdkDisposable);
+        }
+        Disposer.register(project, sdkDisposable);
+        List<Sdk> pythonSdks = PythonSdkUtil.getAllSdks();
+        for (Sdk pythonSdk : pythonSdks) {
+            getLog().debug("Found Python SDK: " + pythonSdk.getName());
+            PyPackageUtil.runOnChangeUnderInterpreterPaths(pythonSdk, sdkDisposable, this::asyncScanAndUpdateResults);
+        }
+        this.pythonSdks = pythonSdks;
+    }
+
+    /**
+     * Create a root node of all Python SDKs.
+     *
+     * @return root node of all Python SDKs.
+     */
+    private DependencyTree createRootNode() {
+        DependencyTree rootNode = new DependencyTree(project.getName());
+        GeneralInfo generalInfo = new GeneralInfo().artifactId(project.getName()).path(Utils.getProjectBasePath(project).toString()).pkgType("pypi");
+        rootNode.setGeneralInfo(generalInfo);
+        rootNode.setScopes(Sets.newHashSet(new Scope()));
+        return rootNode;
+    }
+
+    /**
+     * Create a dependency tree for a given Python SDK.
+     *
+     * @param pythonSdk - The python SDK
+     * @return dependency tree created for a given Python SDK.
+     */
+    private DependencyTree createSdkDependencyTree(Sdk pythonSdk) throws ExecutionException {
+        // Retrieve all Pypi packages
+        PyPackageManager packageManager = PyPipEnvPackageManager.getInstance(pythonSdk);
+        List<PyPackage> packages = packageManager.refreshAndGetPackages(true);
+        getLog().debug(CollectionUtils.size(packages) + " Pypi packages found in SDK " + pythonSdk.getName());
+
+        // Create root SDK node
+        DependencyTree sdkNode = new DependencyTree(pythonSdk.getName());
+        initDependencyNode(sdkNode, pythonSdk.getName(), pythonSdk.getVersionString(), pythonSdk.getHomePath(), "Python SDK");
+
+        // Create dependency mapping
+        Map<String, PyPackage> dependencyMapping = new HashMap<>();
+        for (PyPackage pyPackage : packages) {
+            dependencyMapping.put(pyPackage.getName().toLowerCase(), pyPackage);
+        }
+
+        // Populate dependency tree
+        Collection<PyPackage> values = dependencyMapping.values();
+        Set<String> allDependencies = values.parallelStream()
+                .map(PyPackage::getRequirements)
+                .flatMap(Collection::stream)
+                .map(PyRequirement::getName)
+                .map(String::toLowerCase)
+                .collect(Collectors.toSet());
+        for (PyPackage pyPackage : values) {
+            // If pyPackage contained in one of the dependencies, we conclude it is transitive dependency.
```

Comment:
If pyPackage contained... --> If a pyPackage is contained...
...conclude it is transitive dependency. --> ...conclude it is a transitive dependency.